pkg/controller/restore_controller_test.go

```diff
@@ -450,6 +485,171 @@ func TestProcessRestore(t *testing.T) {
+			Name:      "restore-1",
+		},
+		Spec: api.RestoreSpec{
+			ScheduleName: "schedule-1",
+		},
+	}
+
+	// no backups created from the schedule: fail validation
+	require.NoError(t, sharedInformers.Ark().V1().Backups().Informer().GetStore().Add(arktest.
+		NewTestBackup().
+		WithName("backup-1").
+		WithLabel("ark-schedule", "non-matching-schedule").
+		WithPhase(api.BackupPhaseCompleted).
+		Backup,
+	))
+
+	errs := c.completeAndValidate(restore)
+	assert.Equal(t, []string{"No backups found for schedule"}, errs)
+	assert.Empty(t, restore.Spec.BackupName)
+
+	// no completed backups created from the schedule: fail validation
+	require.NoError(t, sharedInformers.Ark().V1().Backups().Informer().GetStore().Add(arktest.
+		NewTestBackup().
+		WithName("backup-2").
+		WithLabel("ark-schedule", "schedule-1").
+		WithPhase(api.BackupPhaseInProgress).
+		Backup,
+	))
+
+	errs = c.completeAndValidate(restore)
+	assert.Equal(t, []string{"No completed backups found for schedule"}, errs)
+	assert.Empty(t, restore.Spec.BackupName)
+
+	// multiple completed backups created from the schedule: use most recent
+	// (defined as last in alphabetical order)
+	require.NoError(t, sharedInformers.Ark().V1().Backups().Informer().GetStore().Add(arktest.
+		NewTestBackup().
+		WithName("a").
+		WithLabel("ark-schedule", "schedule-1").
+		WithPhase(api.BackupPhaseCompleted).
+		Backup,
+	))
+	require.NoError(t, sharedInformers.Ark().V1().Backups().Informer().GetStore().Add(arktest.
+		NewTestBackup().
+		WithName("b").
+		WithLabel("ark-schedule", "schedule-1").
+		WithPhase(api.BackupPhaseCompleted).
+		Backup,
+	))
+
+	errs = c.completeAndValidate(restore)
+	assert.Nil(t, errs)
+	assert.Equal(t, "b", restore.Spec.BackupName)
+}
+
+func TestBackupXorScheduleProvided(t *testing.T) {
+	r := &api.Restore{}
+	assert.False(t, backupXorScheduleProvided(r))
+
+	r.Spec.BackupName = "backup-1"
+	r.Spec.ScheduleName = "schedule-1"
+	assert.False(t, backupXorScheduleProvided(r))
+
+	r.Spec.BackupName = "backup-1"
+	r.Spec.ScheduleName = ""
+	assert.True(t, backupXorScheduleProvided(r))
+
+	r.Spec.BackupName = ""
+	r.Spec.ScheduleName = "schedule-1"
+	assert.True(t, backupXorScheduleProvided(r))
+
+}
+
+func TestMostRecentCompletedBackup(t *testing.T) {
+	backups := []*api.Backup{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "a",
+			},
+			Status: api.BackupStatus{
+				Phase: "",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "b",
+			},
+			Status: api.BackupStatus{
+				Phase: api.BackupPhaseNew,
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "c",
+			},
+			Status: api.BackupStatus{
+				Phase: api.BackupPhaseInProgress,
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "d",
+			},
+			Status: api.BackupStatus{
+				Phase: api.BackupPhaseFailedValidation,
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "e",
+			},
+			Status: api.BackupStatus{
+				Phase: api.BackupPhaseFailed,
+			},
+		},
+	}
```

Comment:
Maybe add some cases where there are numerals in the names, too. Or switch to using the Backup.Status.StartTimestamp